runtime/src/commonTest/kotlin/pbandk/json/AnyTest.kt

```diff
@@ -0,0 +1,83 @@
+package pbandk.json
+
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.buildJsonObject
+import kotlinx.serialization.json.put
+import pbandk.Message
+import pbandk.add
+import pbandk.pack
+import pbandk.testpb.TestAllTypesProto3
+import pbandk.typeRegistry
+import pbandk.wkt.Any
+import pbandk.wkt.Int32Value
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class AnyTest {
+    private lateinit var jsonConfig: JsonConfig
+
+    private inline fun <T : Message> assertJsonRoundTrip(message: T, expectedJson: JsonObject) {
+        val actualJson = Json.parseToJsonElement(message.encodeToJsonString(jsonConfig))
+        assertEquals(expectedJson, actualJson)
+
+        val actualProto = message.descriptor.messageCompanion.decodeFromJsonString(expectedJson.toString(), jsonConfig)
+        assertEquals(message, actualProto)
+    }
+
+    @BeforeTest
+    fun beforeTest() {
+        jsonConfig = JsonConfig.DEFAULT.copy(
+            typeRegistry = typeRegistry {
+                add(TestAllTypesProto3)
```

Comment:
We could also include a test where an empty typeRegistry is provided to ascertain that encode/decode fails.

Reply:
Good idea 👍🏼  Done.